### PR TITLE
add definition for useRoutes

### PR DESCRIPTION
**Pull request by superhawk610**
v6 introduces a new `useRoutes` hook - this PR adapts the definition from the source.

### CODE REVIEW

File: types/react-router/index.d.ts [hunk 1]
```diff
@@ -164,3 +165,36 @@ export function useRouteMatch<Params extends { [K in keyof Params]?: string } =
...
+    children: RouteConfig[];
+}
+
+export interface RouteRenderWithChildrenConfig {
+    path: string;
+    element: React.ReactNode;
+}
+
+export interface RouteRedirectConfig {
+    path: string;
```
Comment (murbanowicz) [inline types/react-router/index.d.ts]: Path is common part of all configs. Please make it as a base Config and make others extending it.

File: types/react-router/index.d.ts [hunk 2]
```diff
@@ -164,3 +165,36 @@ export function useRouteMatch<Params extends { [K in keyof Params]?: string } =
...
+ * elements in the tree must render an <Outlet> to render their child route's
+ * element.
+ *
+ * Route objects may take one of 2 forms:
+ *
+ * - `{ path, element?, children }`
+ * - `{ path, redirectTo }`
+ * 
+ * If no `element` is provided, `<Outlet />` will be used implicitly.
+ */
```
Comment (murbanowicz) [inline types/react-router/index.d.ts]: I appreciate the explanation, but I would say this is not a place for comments, so I would remove it.

**Review by murbanowicz — CHANGES_REQUESTED** [review]
Thank you so much for helping to make it happen! I highly appreciate any help with that.
Next step is to manage braking changes in a way which will work with dependencies checks on DT.

File: types/react-router/index.d.ts (hunk 2 above)
Comment (superhawk610) [reply types/react-router/index.d.ts]: I wish these docs were included on all functions as they enable really nicely formatted hover docs in editors that support LSPs.

image

If we add them it should be for all, not just this function, so I'm cool with removing them for now.

FWIW it looks like the `react-router` devs agree - https://github.com/ReactTraining/react-router/blob/723fb3b2b95f840981d7da4af3bd625e01997399/packages/react-router/index.js#L468

**Review by murbanowicz — APPROVED** [review]
Thank you

**Pull request merged**